rackcity/views/power_views.py

```diff
@@ -140,7 +140,16 @@ def pdu_power_on(request):
             html.text, power_connections[connection]["port_number"]
         )[0]
         if power_status != "ON":
-            toggle_pdu_power(asset, connection, "on")
+            try:
```

Comment:
The issue was that the actual power switching occurs in this `toggle_pdu_power` method, but rather than propagating the `ConnectionError` exception, I was just returning it from that method, so the actual endpoint method never throws an error when attempting to toggle power. I changed the `toggle_pdu_power` call for power switching/cycling methods to be in a try-except block and also changed the method itself to raise an error rather than returning it